fiksi/src/variable_map.rs

```diff
@@ -51,3 +71,25 @@ impl VariableMap<'_> {
         }
     }
 }
+
+/// A [`VariableMap`], mapping indices directly to a slice.
+///
+/// All variables are considered free.
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct IdentityVariableMap<'s> {
+    pub(crate) variable_values: &'s [f64],
+}
```

Comment:
Simplified this a bit, as in the case of all variables being fixed, we know at the site of use to just not call the gradient-calculating functions.